test/benchmarkTests/index.ts

```diff
@@ -0,0 +1,134 @@
+import { BaseEvent, DocfxBuildFinished, DocfxRestoreFinished, RefreshCredential } from "../../src/common/loggingEvents";
+import { createSandbox } from 'sinon';
+import { EventType } from "../../src/common/eventType";
+import * as converter from 'number-to-words';
+import { formatDuration, getRepositoryInfoFromLocalFolder } from "../../src/utils/utils";
+import { ReportItem, BenchmarkReport } from "../../src/shared";
+
+const testRound = 3;
+
+export function run(): Promise<void> {
+    // tslint:disable-next-line: promise-must-complete
+    return new Promise(async (resolve, reject) => {
+        let buildCount = 0;
+        let sinon = createSandbox();
+        let start: number;
+        let restoreStart: number;
+        let buildStart: number;
+        let restoreDuration = 0;
+        let buildDuration = 0;
+        let totalDuration = 0;
+        let workspace = vscode.workspace.workspaceFolders[0];
+        let [url, branch, commit] = await getRepositoryInfoFromLocalFolder(workspace.uri.fsPath);
+        let report = <BenchmarkReport>{
+            name: workspace.name,
+            url,
+            branch,
+            commit,
+            items: []
+        };
+        const extension = await ensureExtensionActivatedAndInitializationFinished();
+        const { eventStream, keyChain } = extension.exports;
+
+        let subscription = eventStream.subscribe((event: BaseEvent) => {
+            switch (event.type) {
+                case EventType.CredentialRetrieveFromLocalCredentialManager:
+                    console.log(`Trigger build to restore all the dependency...`);
+                    triggerBuild();
+                    break;
+                case EventType.BuildInstantReleased:
+                    handleBuildInstantReleased();
+                    break;
+                case EventType.DocfxRestoreStarted:
+                    restoreStart = Date.now();
+                    break;
+                case EventType.DocfxRestoreFinished:
+                    handleDocfxRestoreFinished(<DocfxRestoreFinished>event);
+                    break;
+                case EventType.DocfxBuildStarted:
+                    buildStart = Date.now();
+                    break;
+                case EventType.DocfxBuildFinished:
+                    handleDocfxBuildFinished(<DocfxBuildFinished>event);
+                    break;
+                case EventType.BuildJobSucceeded:
+                    totalDuration = Date.now() - start;
+                    console.log(`  Build job finished in ${formatDuration(buildDuration)}`);
+                    break;
+                case EventType.BuildTriggerFailed:
+                case EventType.BuildJobFailed:
+                    exitTest(1);
+            }
+        });
+
+        // Prepare
+        setupAvailableMockKeyChain(sinon, keyChain);
+
+        // Refresh the credential
+        eventStream.post(new RefreshCredential());
```

Comment:
>RefreshCredential [](start = 29, length = 17)

Can build trigger be more explicit? It's hard to infer that this triggers a build.

Reply:
this won't trigger a build, the build is triggered in line 41. maybe  I can add some comments here like:
```
// Refresh the credential, after credential refreshed, trigger the build
```

---
In reply to: [368476963](https://github.com/docascode/vscode-docs-build/pull/16#discussion_r368476963) [](ancestors = 368476963)